test/helpers/UnitTestResourceLoader.js

```diff
@@ -24,13 +24,7 @@ module.exports = class UnitTestResourceLoader {
     callback(data);
   }
 
-  loadSpriteSheet(baseName, frameData) {
-    return new Promise(resolve => {
-      const image = new Image();
-      const sheet = this.p5_.loadSpriteSheet(image, frameData);
-      setTimeout(() => {
-        resolve(sheet);
-      }, 0);
-    });
+  async loadSpriteSheet(baseName, frameData) {
+    return this.p5_.loadSpriteSheet(new Image(), frameData);
```

Comment:
@islemaster this change allows you to avoid `tape-async` - but we're also not really testing the asynchronous behavior anymore. Is that a good thing?

Reply:
Tests were passing without `tape-async` before I removed the `setTimeout` here (see Travis history on my commits).  I also suspect we are still genuinely testing async behavior because most promise implementations won't resolve synchronously.

Either way, in a perfect world promise resolution would be a black-box to our code - it shouldn't care whether a promise resolves synchronously or not.  I propose that it's nice to have coverage of both case (near-synchronous here in unit tests, and definitely async in our integration tests) to enforce that.

Reply:
ok with me. several times while building this, I tried removing `tape-async` everywhere, but would see "text exited without end()" or something to that effect. But if it is passing now without it, we can leave it out.

Reply:
Ah - we actually had a test failure get into master branch a while back that looked like that, as a result of a test not cleaning up `sinon.useFakeTimers()`.  It's possible you were running into that.